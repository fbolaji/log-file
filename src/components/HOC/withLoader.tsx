import React, { Component, PropsWithChildren} from "react";

type IProps = {
    loading: boolean;
}

export const withLoader = (Component: PropsWithChildren<div>) => ({ loading, ...props}: IProps) => {
  return loading ? (
    <div data-testid="test-loader" className="loader-overlay">
      <div className="loader-circle-wrap">
        <div className="loader-circle" />
      </div>
    </div>
  ) : (<Component {...props} />)
};

export default withLoader;
