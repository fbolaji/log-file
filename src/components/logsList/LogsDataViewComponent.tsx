import React from 'react';
import BootstrapTable from 'react-bootstrap-table-next';
import paginationFactory from 'react-bootstrap-table2-paginator';

type IProps = {
    list: [
        {
            id: number;
            datetime: string;
            severity: string;
            message: string;
        }
    ];
};

export const LogsDataViewComponent = (props: IProps) => {
    const columns = [{
        dataField: 'id',
        text: '#'
    }, {
        dataField: 'date',
        text: 'dateTime'
    }, {
        dataField: 'severity',
        text: 'severity'
    }, {
        dataField: 'message',
        text: 'message'
    }];
    const { list } = props;
    const options = {
        onSizePerPageChange: (sizePerPage: number, page: number) => {
            console.log('Size per page change!!!');
            console.log('Newest size per page:' + sizePerPage);
            console.log('Newest page:' + page);
        },
        onPageChange: (page: number, sizePerPage: number) => {
            console.log('Page change!!!');
            console.log('Newest size per page:' + sizePerPage);
            console.log('Newest page:' + page);
        }
    };

    // @ts-ignore
    return (

        <BootstrapTable
            striped
            bordered
            hover
            keyField='id'
            data={ list }
            columns={ columns }
            pagination={ paginationFactory(options) }
        />
    )
};

export default LogsDataViewComponent;
