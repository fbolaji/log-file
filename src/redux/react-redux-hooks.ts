import { Dispatch } from "react";
import {
    useDispatch as originalUseDispatch
} from 'react-redux';

import { ISetLogfileAction} from "./actions/logfile.action";

export const useDispatch = () => originalUseDispatch<Dispatch<ISetLogfileAction>>();