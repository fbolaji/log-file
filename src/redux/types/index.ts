export const SAVE_LOGS = 'SAVE_LOGS';