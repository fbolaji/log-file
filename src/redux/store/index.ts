import { applyMiddleware, createStore, compose } from 'redux';
import thunk from 'redux-thunk';
import rootReducer from '../reducers/rootReducer';
import { composeWithDevTools } from 'redux-devtools-extension';

const composeEnhancers = compose;

const store = createStore(
    rootReducer,
    composeWithDevTools(composeEnhancers(applyMiddleware(thunk)))
);

export default store;