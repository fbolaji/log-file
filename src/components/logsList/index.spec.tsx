import React from 'react';
import { render } from '@testing-library/react';

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'
import { Provider } from "react-redux";
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import * as ReactReduxHooks from '../../redux/react-redux-hooks';
import { withLoader } from '../HOC/withLoader';

import LogsListComponent from '../logsList/index';

import configureMockStore from 'redux-mock-store';
import LogsDataViewComponent from "./LogsDataViewComponent";

configure({ adapter: new Adapter() });

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

const RenderLogs = withLoader(LogsDataViewComponent);


describe('LogsListComponent', () => {
    let useEffect: any;
    let store: any;

    const mockUseEffect = () => {
        useEffect.mockImplementationOnce(f => f());
    };

    const response = [
        {
            id: 1595610796967,
            datetime: '2020-07-24T17:13:16.909Z',
            severity: 'INFO',
            message: 'Some INFO message'
        },
        {
            id: 1595610806034,
            datetime: '2020-07-24T17:13:25.976Z',
            severity: 'WARNING',
            message: 'Some WARNING message'
        },
        {
            id: 1595610815087,
            datetime: '2020-07-24T17:13:35.054Z',
            severity: 'INFO',
            message: 'Some INFO message'
        },
        {
            id: 1595610824112,
            datetime: '2020-07-24T17:13:44.101Z',
            severity: 'INFO',
            message: 'Some INFO message'
        },
        {
            id: 1595610833176,
            datetime: '2020-07-24T17:13:53.153Z',
            severity: 'WARNING',
            message: 'Some WARNING message'
        },
        {
            id: 1595610842236,
            datetime: '2020-07-24T17:14:02.220Z',
            severity: 'WARNING',
            message: 'Some WARNING message'
        },
        {
            id: 1595610851313,
            datetime: '2020-07-24T17:14:11.266Z',
            severity: 'ERROR',
            message: 'Some ERROR message'
        },
        {
            id: 1595610860351,
            datetime: '2020-07-24T17:14:20.332Z',
            severity: 'ERROR',
            message: 'Some ERROR message'
        },
        {
            id: 1595610869450,
            datetime: '2020-07-24T17:14:29.397Z',
            severity: 'INFO',
            message: 'Some INFO message'
        },
        {
            id: 1595610878453,
            datetime: '2020-07-24T17:14:38.443Z',
            severity: 'WARNING',
            message: 'Some WARNING message'
        },
        {
            id: 1595610887526,
            datetime: '2020-07-24T17:14:47.500Z',
            severity: 'ERROR',
            message: 'Some ERROR message'
        }
    ];

    beforeEach(() => {
        /* mocking store */
        store = configureStore([thunk])({
            logs: [],
        });

        /* mocking useEffect */
        useEffect = jest.spyOn(React, "useEffect");
        mockUseEffect(); // 2 times
        mockUseEffect(); //

        /* mocking useDispatch on our mock store  */
        jest
            .spyOn(ReactReduxHooks, "useDispatch")
            .mockImplementation(() => store.dispatch);
        /* shallow rendering */

    });

    describe("on mount", () => {
        it('should match snapshot', () => {
            const { container } = render(<Provider store={store} ><LogsListComponent /></Provider>);
            expect(container).toMatchSnapshot();
        });
        it('should render', () => {
            const { container } = render(<RenderLogs loading={true} list={response} />);
            expect(container).toBeInTheDocument();
        });
        it('should render Chart component', () => {
            const { getByTestId } = render(<Provider store={store} ><LogsListComponent /></Provider>);

            expect(getByTestId('logStatistic')).toBeTruthy();
        })
    });
});
