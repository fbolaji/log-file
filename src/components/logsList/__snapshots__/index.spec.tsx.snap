// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogsListComponent on mount should match snapshot 1`] = `
<div>
  <div
    class="mt-2 row"
  >
    <div
      class="col-md-7 col-12 order-md-1 order-2"
    >
      <div
        class="react-bootstrap-table"
      >
        <table
          class="table table-striped table-hover table-bordered logsTable"
        >
          <thead>
            <tr>
              <th
                tabindex="0"
              >
                #
              </th>
              <th
                tabindex="0"
              >
                DateTime
              </th>
              <th
                tabindex="0"
              >
                Severity
              </th>
              <th
                tabindex="0"
              >
                Message
              </th>
            </tr>
          </thead>
          <tbody>
            <tr
              style="background-color: rgb(52, 58, 64); color: rgb(204, 204, 204);"
            >
              <td>
                1595610796967
              </td>
              <td />
              <td>
                INFO
              </td>
              <td>
                Some INFO message
              </td>
            </tr>
            <tr
              style="background-color: rgb(33, 37, 41); color: rgb(204, 204, 204);"
            >
              <td>
                1595610806034
              </td>
              <td />
              <td>
                WARNING
              </td>
              <td>
                Some WARNING message
              </td>
            </tr>
            <tr
              style="background-color: rgb(52, 58, 64); color: rgb(204, 204, 204);"
            >
              <td>
                1595610815087
              </td>
              <td />
              <td>
                INFO
              </td>
              <td>
                Some INFO message
              </td>
            </tr>
            <tr
              style="background-color: rgb(33, 37, 41); color: rgb(204, 204, 204);"
            >
              <td>
                1595610824112
              </td>
              <td />
              <td>
                INFO
              </td>
              <td>
                Some INFO message
              </td>
            </tr>
            <tr
              style="background-color: rgb(52, 58, 64); color: rgb(204, 204, 204);"
            >
              <td>
                1595610833176
              </td>
              <td />
              <td>
                WARNING
              </td>
              <td>
                Some WARNING message
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div
        class="row react-bootstrap-table-pagination"
      >
        <div
          class="col-md-6 col-xs-6 col-sm-6 col-lg-6"
        />
        <div
          class="react-bootstrap-table-pagination-list col-md-6 col-xs-6 col-sm-6 col-lg-6"
        >
          <ul
            class="pagination react-bootstrap-table-page-btns-ul"
          >
            <li
              class="active page-item"
              title="1"
            >
              <a
                class="page-link"
                href="#"
              >
                1
              </a>
            </li>
            <li
              class="page-item"
              title="2"
            >
              <a
                class="page-link"
                href="#"
              >
                2
              </a>
            </li>
            <li
              class="page-item"
              title="3"
            >
              <a
                class="page-link"
                href="#"
              >
                3
              </a>
            </li>
            <li
              class="page-item"
              title="next page"
            >
              <a
                class="page-link"
                href="#"
              >
                &gt;
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      class="col-md-5 col-12 order-md-2 order-1"
    >
      <aside
        data-testid="logStatistic"
      >
        <div>
          Loading Chart
        </div>
      </aside>
    </div>
  </div>
</div>
`;
